src/scopes/activityReport/goalName.ts

```diff
@@ -0,0 +1,22 @@
+import { Op } from 'sequelize';
+import { filterAssociation, argsIncludeExclude } from './utils';
+
+export function withGoalName(searchText: string[]) {
+  const search = [`%${searchText.map((st: string) => st.toLowerCase())}%`];
+
+  return {
+    [Op.and]: [
+      filterAssociation(argsIncludeExclude(true), search, false, 'LIKE'),
```

Comment:
this code path exposes a potential sql injection vector, while i know its not only exposed by this PR it is continuing to use it. Should it be addressed as part of this JIRA or a new one?

Reply:
I will add sequelize.escape where necessary to the utils file

Reply:
upon further investigation, there is no injection vulnerability. Search terms are passed to the reportInSubQuery function, which escapes the search terms

```js
function reportInSubQuery(baseQuery, searchTerms, operator, comparator) {
  return searchTerms.map((term) => sequelize.literal(`"ActivityReport"."id" ${operator} (${baseQuery} ${comparator} ${sequelize.escape(String(term).trim())})`));
}